sim/druid/item_sets_pve_phase_5.go

```diff
@@ -295,32 +264,61 @@ func (druid *Druid) applyT2Guardian2PBonus() {
 
 // Your Mangle(Bear), Swipe(Bear), Maul, and Lacerate abilities gain 5% increased critical strike chance against targets afflicted by your Lacerate.
 func (druid *Druid) applyT2Guardian4PBonus() {
-	druid.FuryOfStormrageCritRatingBonus = 5 * core.SpellCritRatingPerCritChance
+	if !druid.HasRune(proto.DruidRune_RuneLegsLacerate) {
+		return
+	}
+
+	label := "S03 - Item - T2 - Druid - Guardian 4P Bonus"
+	if druid.HasAura(label) {
+		return
+	}
+
+	druid.RegisterAura(core.Aura{
+		Label: label,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			for _, spell := range []*DruidSpell{druid.MangleBear, druid.SwipeBear, druid.Maul, druid.Lacerate} {
+				if spell == nil {
+					continue
+				}
+
+				oldApplyEffects := spell.ApplyEffects
+				spell.ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+					bonusCrit := 0.0
+					if druid.LacerateBleed.Dot(target).GetStacks() > 0 {
+						bonusCrit = 5 * core.CritRatingPerCritChance
+					}
+
+					spell.BonusCritRating += bonusCrit
+					oldApplyEffects(sim, target, spell)
+					spell.BonusCritRating -= bonusCrit
+				}
+			}
+		},
+	})
 }
 
 // Your Swipe now spreads your Lacerate from your primary target to other targets it strikes.
 func (druid *Druid) applyT2Guardian6PBonus() {
-	// if druid.Env.GetNumTargets() == 1 {
-	// 	return
-	// }
-
-	// targetCount := core.TernaryInt32(druid.HasRune(proto.DruidRune_RuneCloakImprovedSwipe), 10, 3)
-	// numHits := min(targetCount, druid.Env.GetNumTargets())
-	// results := make([]*core.SpellResult, numHits)
-
-	// core.MakePermanent(druid.RegisterAura(core.Aura{
-	// 	Label: "S03 - Item - T2 - Druid - Guardian 6P Bonus",
-	// 	OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-	// 		if spell.SpellCode == SpellCode_DruidSwipeBear {
-	// 			// TODO: Troubleshoot - Saeyon
-	// 			for _, cleaveMob := range results {
-	// 				druid.LacerateBleed.Cast(sim, cleaveMob.Target)
-	// 				druid.LacerateBleed.Dot(cleaveMob.Target).SetStacks(sim, druid.LacerateBleed.Dot(result.Target).GetStacks())
-	//              // Bleed duration needs to match the original DOT duration
-	// 			}
-	// 		}
-	// 	},
-	// }))
+	if !druid.HasRune(proto.DruidRune_RuneLegsLacerate) {
+		return
+	}
+	druid.FuryOfStormrageLacerateSpread = true
+	core.MakePermanent(druid.RegisterAura(core.Aura{
+		Label: "S03 - Item - T2 - Druid - Guardian 6P Bonus",
```

Comment:
This is missing the `HasAura` check to make sure the bonus doesn't get duplicated when using the gear + shoulder run